addon/components/plugins/text-style/color.ts

```diff
@@ -0,0 +1,97 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { SayController } from '@lblod/ember-rdfa-editor';
+import { clearColor, setColor } from '@lblod/ember-rdfa-editor/plugins/color';
+import { paintCycleHappened } from '@lblod/ember-rdfa-editor/utils/_private/editor-utils';
+import { modifier } from 'ember-modifier';
+import { TextSelection } from 'prosemirror-state';
+
+type Args = {
+  controller: SayController;
+  defaultColor: string;
+};
+
+export default class ColorMenu extends Component<Args> {
+  dropdownButton?: HTMLElement;
+
+  setupDropdownButton = modifier(
+    (element: HTMLElement) => {
+      this.dropdownButton = element;
+    },
+    { eager: false }
+  );
+  @tracked dropdownOpen = false;
+
+  get controller() {
+    return this.args.controller;
+  }
+
+  @action
+  toggleDropdown() {
+    this.dropdownOpen = !this.dropdownOpen;
+  }
+
+  get currentColor() {
+    if (this.controller) {
+      const state = this.controller.mainEditorState;
+      const { selection } = state;
+      const marks = state.storedMarks;
+      const markType = state.schema.marks.color;
+
+      if (marks?.length) {
+        const markSet = markType.isInSet(marks);
+        const color = markSet?.attrs?.color as string;
+
+        return color || 'black';
+      } else {
+        let node;
+
+        if (selection instanceof TextSelection) {
+          node = this.controller.mainEditorState.doc.nodeAt(
+            selection.$from.pos
+          );
+        }
+
+        if (node === null && selection.$from.parentOffset > 0) {
+          node = this.controller.mainEditorState.doc.nodeAt(
+            selection.$from.pos - 1
+          );
+        }
+
+        const markSet = markType.isInSet(node?.marks || []);
+        const color = markSet?.attrs?.color as string;
```

Comment:
I think you can use `selection.$from.marks()` in order to get the marks which are active at the beginning of the selection.